### ISSUE TITLE
[HPOS] Move HPOS from experimental to stable

### ISSUE THREAD

**Issue by vedanshujain** [woothemes/woocommerce#36377]
Depends on #36376.

Now that we have e2e and API tests enabled for HPOS and another issue for enabling unit tests as well (#36376), it's time to mark HPOS as stable.

This issue scopes removing the developer warning from the feature page and CLIRunner.
